src/components/SearchForm.js

```diff
@@ -0,0 +1,73 @@
+import { useContext, useState } from 'react';
+import { GoogleMapsContext } from '../context/GoogleMapsContext';
+import { MapCenterContext } from '../context/MapCenterContext';
+import planetSearch from '../assets/search-location-solid.svg';
+
+export const SearchForm = () => {
+  const [query, setQuery] = useState('');
+  const valueGoogleMapsAPI = useContext(GoogleMapsContext);
+  const valueMapCenter = useContext(MapCenterContext);
+  const { map, maps } = valueGoogleMapsAPI;
+  const { setNewCenterMap } = valueMapCenter;
+
+  const handleQuery = (e) => setQuery(e.target.value);
+
+  const handleSubmit = (e) => {
+    e.preventDefault();
+
+    var request = {
+      query,
+      fields: ['name', 'geometry'],
+    };
+
+    var service = new maps.places.PlacesService(map);
+
+    service.findPlaceFromQuery(request, function (results, status) {
+      if (status === maps.places.PlacesServiceStatus.REQUEST_DENIED) {
+        alert(
+          'We are so sorry! Places API is not enabled. Please try again later.',
+        );
+      }
+      if (status === maps.places.PlacesServiceStatus.ZERO_RESULTS) {
+        alert('No matching results');
+      }
+      if (status === maps.places.PlacesServiceStatus.OK) {
+        const firstResult = results[0];
+        const coordinates = firstResult?.geometry?.location;
+        const lat = coordinates.lat();
+        const lng = coordinates.lng();
+        const newMapCenter = { lat, lng };
+        setNewCenterMap(newMapCenter);
+      }
+    });
+  };
+
+  return (
+    <>
+      {maps !== null ? (
```

Comment:
Not sure why this check is here

Reply:
Thanks for your question, Katia!

We are trying to control whether Google Maps API is enabled, exceptions related to null values, and the map has already loaded.